### PR TITLE
feat(remix-vercel): add support for `@vercel/node` v2

### PR DESCRIPTION
**Pull request by mcansh**
Closes: #

- [ ] Docs
- [ ] Tests

Testing Strategy:

### CODE REVIEW

**Comment by MichaelDeBoey** [comment]
We already have #3412

**Comment by MichaelDeBoey** [comment]
@mcansh We have permission from @styfle to not wait for approval of his employer and re-open & merge this one

**Review by MichaelDeBoey — APPROVED** [review]

**Comment by mcansh** [comment]
> @mcansh Seems like build & tests are failing. Probably need to re-generate `yarn.lock`

caused by our test using an internal vercel/node module, looking into it

**Comment by MichaelDeBoey** [comment]
Seems like https://github.com/vercel/vercel/issues/7931 is causing the build to fail 😕

**Comment by mcansh** [comment]
looks like `@vercel/node` uses `@vercel/build-utils` but it's not in its package.json 👀

**Comment by MichaelDeBoey** [comment]
Created https://github.com/vercel/vercel/pull/7934 to fix it

**Comment by styfle** [comment]
How about installing `@vercel/build-utils` to your devDependencies?

**Comment by MichaelDeBoey** [comment]
@styfle It's used by `@vercel/node`, so that package should have it in `dependencies`

File: packages/remix-vercel/package.json
Comment (MichaelDeBoey) [inline packages/remix-vercel/package.json:19]:
```suggestion
    "@vercel/node": "^2.2.0",
```

File: packages/remix-vercel/package.json
Comment (MichaelDeBoey) [inline packages/remix-vercel/package.json:25]: Not sure about this tbh 🤔
Technically `2.0` if fine if people install `@vercel/build-utils` themselves

```suggestion
    "@vercel/node": "^1.8.3 || ^2.2.0"
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]
@mcansh `@vercel/node@2.2.0` is [released](https://github.com/vercel/vercel/releases/tag/%40vercel%2Fnode%402.2.0) now 🥳🎉

File: packages/remix-vercel/package.json
Comment (mcansh) [reply packages/remix-vercel/package.json:25]: i think 2.2 is fine as it doesn't require the extra dep in user land

File: packages/remix-vercel/package.json
Comment (MichaelDeBoey) [reply packages/remix-vercel/package.json]: @mcansh This means we'll have to re-generate `yarn.lock`

Comment (mcansh) [reply packages/remix-vercel/package.json]: 👍 i was AFK when i got the earlier notification, doing that now

**Comment by MichaelDeBoey** [comment]
@styfle It seems like `@vercel/build-utils` is missing `@vercel/frameworks` & `@vercel/routing-utils` in it's `dependencies` list.
https://github.com/remix-run/remix/runs/6924983939?check_suite_focus=true#step:6:124

```sh
Error: node_modules/@vercel/build-utils/dist/detect-builders.d.ts(1,23): error TS2307: Cannot find module '@vercel/routing-utils' or its corresponding type declarations.
Error: node_modules/@vercel/build-utils/dist/detect-framework.d.ts(1,27): error TS2307: Cannot find module '@vercel/frameworks' or its corresponding type declarations.
Error: node_modules/@vercel/build-utils/dist/workspaces/workspace-managers.d.ts(1,32): error TS2307: Cannot find module '@vercel/frameworks' or its corresponding type declarations.
Error: node_modules/@vercel/build-utils/dist/monorepos/monorepo-managers.d.ts(1,32): error TS2307: Cannot find module '@vercel/frameworks' or its corresponding type declarations.
```

I however think it would be better to remove `@vercel/build-utils` from `@vercel/node` instead tbh.

**Comment by MichaelDeBoey** [comment]
Waiting for https://github.com/vercel/vercel/issues/7951#issuecomment-1158080552 to be resolved

**Comment by MichaelDeBoey** [comment]
@styfle We've reverted changes to `templates/vercel` for now and we'll wait until everything is fixed in `@vercel/node` before we add support for v2

File: packages/remix-vercel/package.json
Comment (MichaelDeBoey) [inline packages/remix-vercel/package.json:19]:
```suggestion
    "@vercel/node": "^2.3.3",
```

File: packages/remix-vercel/package.json
Comment (MichaelDeBoey) [inline packages/remix-vercel/package.json:25]:
```suggestion
    "@vercel/node": "^1.8.3 || ^2.3.3"
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]
@mcansh `@vercel/node@2.3.3` is [released](https://github.com/vercel/vercel/releases/tag/%40vercel%2Fnode%402.3.3) now 🥳🎉

Don't forget to update `yarn.lock` as well.

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]
@mcansh `@vercel/node@2.4.0` is [released](https://github.com/vercel/vercel/releases/tag/%40vercel%2Fnode%402.4.0) now 🥳🎉

Don't forget to update `yarn.lock` as well.

**Pull request merged**
